index.js

```diff
@@ -0,0 +1,60 @@
+document.addEventListener('DOMContentLoaded', () => {
+// console.log("dom")
+setupDroplist();
+fetchMovieList()
+})
+// add list to drop list
+const setupDroplist = ()=>{
+  const getMovieList = document.querySelector('#droplist')
+  getMovieList.addEventListener('click', fetchMovieList)
+  console.log( "click")
+}
+
+const fetchMovieList =()=>{
+  console.log("fetch was called")
+  fetchAllTitles();
+  // selectedTitle();
+  // fetchYear();
+  // fetchDescription();
+}
+
+const fetchAllTitles=(film)=>{
+  
+  const films =` https://ghibliapi.herokuapp.com/films`
+  // console.log(url)
+  fetch(films)
+  .then(response => response.json())
+  .then(films => {
+    // console.log(films)
+    for (let i = 0; i< films.length; i++) {
+     let titleFilms = films[i].title
+    let options = document.createElement('option');
+    options.value = titleFilms;
+    options.innerText = titleFilms;
+    let select =document.querySelector('#droplist')
+    select.appendChild(options)
+   
+    }
```

Comment:
index.js:29-37. The task that this for loop is performing could have been taken to a separate function say `appendOptions` and be called passing the array of films like so: `appendOptions(films)`. Since this function is being called on click and there is no logic to replace the old list of options, this just keeps appending more and more options to the same `select`. We would need a function like the one I showed in class to "clear" the select options or replace them.